chatmaild/src/chatmaild/filtermail.py

```diff
@@ -23,43 +30,40 @@ def check_encrypted(message):
         return False
     parts_count = 0
     for part in message.iter_parts():
+        # We explicitly check Content-Type of each part later,
+        # but this is to be absolutely sure `get_payload()` returns string and not list.
+        if part.is_multipart():
+            return False
+
         if parts_count == 0:
             if part.get_content_type() != "application/pgp-encrypted":
                 return False
+
+            payload = part.get_payload()
+            if payload.strip() != "Version: 1":
+                return False
         elif parts_count == 1:
             if part.get_content_type() != "application/octet-stream":
                 return False
-        else:
-            return False
-        parts_count += 1
-    return True
-
-
-def check_mdn(message, envelope):
-    if len(envelope.rcpt_tos) != 1:
-        return False
 
-    for name in ["auto-submitted", "chat-version"]:
-        if not message.get(name):
-            return False
-
-    if message.get_content_type() != "multipart/report":
-        return False
+            payload = part.get_payload()
 
-    body = message.get_body()
-    if body.get_content_type() != "text/plain":
-        return False
-
-    if list(body.iter_attachments()) or list(body.iter_parts()):
-        return False
+            prefix = "-----BEGIN PGP MESSAGE-----\r\n\r\n"
+            if not payload.startswith(prefix):
+                return False
+            payload = payload.removeprefix(prefix)
 
-    # even with all mime-structural checks an attacker
-    # could try to abuse the subject or body to contain links or other
-    # annoyance -- we skip on checking subject/body for now as Delta Chat
-    # should evolve to create E2E-encrypted read receipts anyway.
-    # and then MDNs are just encrypted mail and can pass the border
-    # to other instances.
+            suffix = "\r\n-----END PGP MESSAGE-----\r\n\r\n"
+            if not payload.endswith(suffix):
+                return False
+            payload = payload.removesuffix(suffix)
 
+            for line in payload.split("\r\n"):
+                if NONBASE64_REGEX.match(line):
+                    return False
```

Comment:
this could be a multi-line regex match to avoid splitting and iterating the lines.